### ISSUE TITLE
[adjacent-overload-signatures] False positive when static and instance methods share the same name

### ISSUE THREAD

**Issue by Jessidhia** [typescript-eslint/typescript-eslint#169]
**Repro**

```ts
/* eslint @typescript-eslint/adjacent-overload-signatures: error */
class Test {
  static test() {}
  untest() {}
  test() {}
}

```

**Expected Result**

No warning, as despite having the same names they are unrelated. The first `test` is a member of `typeof Test`, the second `test` is a member of `Test`.

**Actual Result**

"All 'test' signatures should be adjacent."

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.1.1` |
| `@typescript-eslint/parser`        | `1.1.1` |
| `TypeScript`                       | `3.2.4` |
| `ESLint`                           | `5.12.1` |
| `node`                             | `11.8.0` |
| `yarn`                             | `1.14.0` |

**Comment by lukyth** [comment]
Can I take a stab at this?
